ui/app/pages/settings/contact-list-tab/edit-contact/edit-contact.component.js

```diff
@@ -0,0 +1,79 @@
+import React, { PureComponent } from 'react'
+import PropTypes from 'prop-types'
+import Identicon from '../../../../components/ui/identicon'
+import { CONTACT_LIST_ROUTE } from '../../../../helpers/constants/routes'
+import { addressSlicer } from '../../../../helpers/utils/util'
+import PageContainerFooter from '../../../../components/ui/page-container/page-container-footer'
+
+export default class EditContact extends PureComponent {
+
+  static contextTypes = {
+    t: PropTypes.func,
+  }
+
+  static propTypes = {
+    addressBook: PropTypes.object,
+    addToAddressBook: PropTypes.func,
+    history: PropTypes.object,
+    match: PropTypes.object,
+  }
+
+  state = {
+    newName: '',
+    newAddress: '',
+  }
+
+   render () {
+    const { t } = this.context
+    const { history, match, addressBook, addToAddressBook } = this.props
+    const address = match.params.id
+    const currentEntry = addressBook[address]
+    const name = currentEntry.name !== '' ? currentEntry.name : addressSlicer(address)
+
+    return (
+      <div className="settings-page__content-row address-book__edit-contact">
+        <div className="settings-page__header address-book__header">
+          <Identicon address={address} diameter={60}/>
+        </div>
+        <div className="address-book__edit-contact__content">
+          <div className="address-book__view-contact__group">
+            <div className="address-book__view-contact__group__label">
+              { t('userName') }
+            </div>
+            <input
```

Comment:
Perhaps use `TextField` component as is done in `ui/app/pages/settings/contact-list-tab/add-contact/add-contact.component.js`